expression/builtin_other.go

```diff
@@ -106,22 +111,71 @@ func (c *inFunctionClass) getFunction(ctx sessionctx.Context, args []Expression)
 // builtinInIntSig see https://dev.mysql.com/doc/refman/5.7/en/comparison-operators.html#function_in
 type builtinInIntSig struct {
 	baseBuiltinFunc
+	hashSet map[int64]bool
+	hasNull bool
+	varArgs []Expression
+}
+
+func (b *builtinInIntSig) buildHashMapForConstArgs(ctx sessionctx.Context) error {
+	b.varArgs = make([]Expression, 0, len(b.args))
+	b.varArgs = append(b.varArgs, b.args[0])
+	b.hashSet = make(map[int64]bool, len(b.args)-1)
+	for i := 1; i < len(b.args); i++ {
+		if b.args[i].ConstItem() {
+			val, isNull, err := b.args[i].EvalInt(ctx, chunk.Row{})
```

Comment:
Could we use `nil` instead of `chunk.Row{}`?

Reply:
the second arg of `EvalInt` need to be the type `chunk.Row` and `chunk.Row` is a struct not a interface